File: Cpp/1629_Slowest_Key.cpp
class Solution {
public:
    char slowestKey(vector<int>& releaseTimes, string keysPressed) {
        char result = keysPressed[0];
        int max_time = releaseTimes[0];
        for(int i=1; i<releaseTimes.size(); ++i){
            int time = releaseTimes[i]-releaseTimes[i-1];
            if(time < max_time) continue;
            else if(time == max_time && keysPressed[i] < result) continue;
            
            max_time = time;
            result = keysPressed[i];
        }
        
        return result;
    }
};

File: Kotlin/1629_Slowest_Key.kt
class Solution {
    fun slowestKey(releaseTimes: IntArray, keysPressed: String): Char {
        var result = keysPressed[0]
        var maxTime = releaseTimes[0]
        for(i in 1 until releaseTimes.size){
            val time = releaseTimes[i] - releaseTimes[i - 1]
            if(time < maxTime) continue
            else if(time == maxTime && keysPressed[i] < result) continue
            
            maxTime = time
            result = keysPressed[i]
        }
        
        return result
    }
}